be/test/storage/persistent_index_test.cpp

```diff
@@ -44,7 +44,10 @@ struct PersistentIndexTestParam {
 class PersistentIndexTest : public testing::TestWithParam<PersistentIndexTestParam> {
 public:
     virtual ~PersistentIndexTest() {}
-    void SetUp() override { config::enable_pindex_compression = GetParam().enable_pindex_compression; }
+    void SetUp() override {
+        config::enable_pindex_compression = GetParam().enable_pindex_compression;
+        config::enable_pindex_read_by_page = true;
```

Comment:
should restore the configvar value in TearDown(), avoid pollute other cases which the behavior might be affected by this configvar.

Reply:
> should restore the configvar value in TearDown(), avoid pollute other cases which the behavior might be affected by this configvar.

This behavior will be set as default behavior in future production env and this pr refrains from altering the default behavior to mitigate potential impacts from minor version upgrades. In actuality, I intend to conduct unit testing with this new behavior, which is why I have opted not to reset the configuration in TearDown().

Reply:
a good practice is to restore the env to exactly the same as the ut enters. if in future the default value is going to be `true`, set it in config.h.

if you want to try it first in ut, set it in be_test.conf or in ut main().

No one knows your intention behind this without any comments.

B.T.W. our CI is running in parallel, each UT suite is running in a separate process, so your intention in this mode doesn't work. Make the code explicitly reflect your design intention.

Reply:
> a good practice is to restore the env to exactly the same as the ut enters. if in future the default value is going to be `true`, set it in config.h.
> 
> if you want to try it first in ut, set it in be_test.conf or in ut main().
> 
> No one knows your intention behind this without any comments.
> 
> B.T.W. our CI is running in parallel, each UT suite is running in a separate process, so your intention in this mode doesn't work. Make the code explicitly reflect your design intention.

A good advice, changed in https://github.com/StarRocks/starrocks/pull/42870/commits/bfa05915a117550d36f0b13b313ec08bb3ea01fb, PTAL again, thx.